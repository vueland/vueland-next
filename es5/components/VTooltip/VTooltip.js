"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.VTooltip = void 0;

require("../../../src/components/VTooltip/VTooltip.scss");

var _vue = require("vue");

var _useToggle2 = require("../../effects/use-toggle");

var _useColors2 = require("../../effects/use-colors");

var _useActivator2 = require("../../effects/use-activator");

var _useTransition = require("../../effects/use-transition");

var _useElevation2 = require("../../effects/use-elevation");

var _usePosition = require("../../effects/use-position");

var _helpers = require("../../helpers");

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var VTooltip = (0, _vue.defineComponent)({
  name: 'v-tooltip',
  props: _objectSpread(_objectSpread({
    openOnHover: {
      type: Boolean,
      "default": true
    },
    color: {
      type: String,
      "default": 'grey lighten-1'
    },
    zIndex: [Number, String],
    maxWidth: [Number, String],
    minWidth: [Number, String],
    modelValue: Boolean
  }, (0, _useElevation2.elevationProps)()), (0, _usePosition.positionProps)()),
  setup: function setup(props, _ref) {
    var slots = _ref.slots;
    var tooltip = (0, _vue.reactive)({});
    var activator = (0, _vue.reactive)({});
    var innerActive = (0, _vue.ref)(false);
    var tooltipRef = (0, _vue.ref)(null);

    var _useToggle = (0, _useToggle2.useToggle)(props),
        isActive = _useToggle.isActive;

    var _useElevation = (0, _useElevation2.useElevation)(props),
        elevationClasses = _useElevation.elevationClasses;

    var _useColors = (0, _useColors2.useColors)(),
        setBackground = _useColors.setBackground;

    var _useActivator = (0, _useActivator2.useActivator)(),
        activatorRef = _useActivator.activatorRef,
        getActivatorSizes = _useActivator.getActivatorSizes,
        genActivatorListeners = _useActivator.genActivatorListeners;

    var listeners = genActivatorListeners(props, innerActive);
    var classes = (0, _vue.computed)(function () {
      return {
        'v-tooltip': true,
        'v-tooltip--top': props.top,
        'v-tooltip--right': props.right,
        'v-tooltip--left': props.left,
        'v-tooltip--bottom': props.bottom
      };
    });
    var contentClasses = (0, _vue.computed)(function () {
      return _objectSpread({
        'v-tooltip__content': true
      }, elevationClasses.value);
    });
    var computeTopPosition = (0, _vue.computed)(function () {
      return (props.top ? activator.top - tooltip.height : props.bottom ? activator.top + activator.height : activator.top + (activator.height - tooltip.height) / 2) + +props.offsetY;
    });
    var computeLeftPosition = (0, _vue.computed)(function () {
      return (props.left ? activator.left - tooltip.width : props.right ? activator.left + activator.width : activator.left + (activator.width - tooltip.width) / 2) + +props.offsetX;
    });
    var styles = (0, _vue.computed)(function () {
      return {
        top: tooltip.top ? (0, _helpers.convertToUnit)(tooltip.top) : '',
        left: tooltip.top ? (0, _helpers.convertToUnit)(tooltip.left) : '',
        maxWidth: !!props.maxWidth ? "".concat(props.maxWidth, "px") : '',
        minWidth: !!props.minWidth ? "".concat(props.minWidth, "px") : '',
        zIndex: props.zIndex
      };
    });

    function genActivator() {
      var slotContent = (0, _vue.renderSlot)(slots, 'activator', {
        on: listeners
      });
      return (0, _vue.h)('div', {
        "class": 'v-tooltip__activator',
        ref: activatorRef
      }, slotContent);
    }

    function genContent() {
      var propsData = {
        "class": contentClasses.value,
        style: styles.value,
        ref: tooltipRef
      };
      return (0, _vue.withDirectives)((0, _vue.h)('span', setBackground(props.color, propsData), slots["default"] && slots["default"]()), [[_vue.vShow, innerActive.value]]);
    }

    function setTooltipPosition() {
      if (tooltipRef.value) {
        tooltip.width = tooltipRef.value.offsetWidth;
        tooltip.height = tooltipRef.value.offsetHeight;
        tooltip.top = computeTopPosition.value;
        tooltip.left = computeLeftPosition.value;
      }
    }

    (0, _vue.watch)(function () {
      return isActive.value;
    }, function (to) {
      return innerActive.value = to;
    }, {
      immediate: true
    });
    (0, _vue.onMounted)(function () {
      (0, _vue.watch)(function () {
        return innerActive.value;
      }, function (to) {
        if (to) {
          var _getActivatorSizes = getActivatorSizes(),
              left = _getActivatorSizes.left,
              top = _getActivatorSizes.top,
              height = _getActivatorSizes.height,
              width = _getActivatorSizes.width;

          activator.left = left;
          activator.top = top;
          activator.height = height;
          activator.width = width;
          tooltip.top = 0;
          tooltip.left = 0;
          requestAnimationFrame(setTooltipPosition);
        }
      }, {
        immediate: true
      });
    });
    return function () {
      var content = (0, _useTransition.useTransition)(genContent(), innerActive.value ? 'scale-in' : 'fade');
      return (0, _vue.h)('div', {
        "class": classes.value
      }, [genActivator(), content]);
    };
  }
});
exports.VTooltip = VTooltip;
//# sourceMappingURL=VTooltip.js.map