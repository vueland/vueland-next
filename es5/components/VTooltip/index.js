"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _VTooltip = require("./VTooltip");

Object.keys(_VTooltip).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (key in exports && exports[key] === _VTooltip[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _VTooltip[key];
    }
  });
});
//# sourceMappingURL=index.js.map