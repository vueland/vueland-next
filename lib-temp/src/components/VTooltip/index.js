export { VTooltip } from './VTooltip';
//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoiaW5kZXguanMiLCJzb3VyY2VSb290IjoiIiwic291cmNlcyI6WyIuLi8uLi8uLi8uLi9zcmMvY29tcG9uZW50cy9WVG9vbHRpcC9pbmRleC50cyJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiQUFBQSxPQUFPLEVBQUUsUUFBUSxFQUFFLE1BQU0sWUFBWSxDQUFBIiwic291cmNlc0NvbnRlbnQiOlsiZXhwb3J0IHsgVlRvb2x0aXAgfSBmcm9tICcuL1ZUb29sdGlwJ1xuIl19