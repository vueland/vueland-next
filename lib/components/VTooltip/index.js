export * from "./VTooltip";
//# sourceMappingURL=index.js.map