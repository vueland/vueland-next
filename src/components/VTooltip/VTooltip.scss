@import "../../styles/scss/modules/global";
@import 'variables';

.v-tooltip {
  position: absolute;
  z-index: 100;
  padding: $tooltip-padding;
  border-radius: 5px;
  color: map-get($white, 'base');
  pointer-events: none;
  max-height: 32px;
  text-align: center;
  font-size: $tooltip-font-size;
  line-height: round($tooltip-font-size * 1.6);

  &__activator {
    display: block;
    height: auto;
    border: none;
  }

  & > * {
    display: block;
    white-space: nowrap;
  }
}
