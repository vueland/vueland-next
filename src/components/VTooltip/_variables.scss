$tooltip-font-size: 14px;
$tooltip-padding: 5px;
$tooltip-background-color: #b6b6b6;
