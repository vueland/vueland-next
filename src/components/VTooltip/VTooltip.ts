// Styles
import './VTooltip.scss'

// Vue API
import {
  h,
  ref,
  reactive,
  watch,
  computed,
  renderSlot,
  withDirectives,
  defineComponent,
  onMounted,
  vShow,
} from 'vue'

// Effects
import { useToggle } from '../../effects/use-toggle'
import { useColors } from '../../effects/use-colors'
import { useActivator } from '../../effects/use-activator'
import { useTransition } from '../../effects/use-transition'
import { elevationProps, useElevation } from '../../effects/use-elevation'
import { positionProps } from '../../effects/use-position'

// Types
import { OffsetSizes } from '../../types'
import { VNode } from 'vue'

// Helpers
import { convertToUnit } from '../../helpers'

const vTooltipProps: any = {
  openOnHover: {
    type: Boolean,
    default: true,
  },
  color: {
    type: String,
    default: 'grey lighten-1',
  },
  zIndex: [Number, String],
  maxWidth: [Number, String],
  minWidth: [Number, String],
  modelValue: Boolean,
  ...elevationProps(),
  ...positionProps(),
}

// TODO fix behavior on window resize if v-model used on component

export const VTooltip = defineComponent({
  name: 'v-tooltip',
  props: vTooltipProps,

  setup(props, { slots }): () => VNode {
    const tooltip = reactive<Partial<OffsetSizes>>({})
    const activator = reactive<Partial<OffsetSizes>>({})

    const innerActive = ref<boolean>(false)
    const tooltipRef = ref<HTMLElement | null>(null)

    const { isActive } = useToggle(props)
    const { elevationClasses } = useElevation(props)
    const { setBackground } = useColors()
    const {
      activatorRef,
      getActivatorSizes,
      genActivatorListeners,
    } = useActivator()

    const listeners = genActivatorListeners(props, innerActive)

    const classes = computed(() => ({
      'v-tooltip': true,
      'v-tooltip--top': props.top,
      'v-tooltip--right': props.right,
      'v-tooltip--left': props.left,
      'v-tooltip--bottom': props.bottom,
    }))

    const contentClasses = computed<Record<string, boolean>>(() => ({
      'v-tooltip__content': true,
      ...elevationClasses.value,
    }))

    const computeTopPosition = computed<number>(() => {
      return (
        (props.top
          ? activator!.top! - tooltip.height!
          : props.bottom
            ? activator.top! + activator.height!
            : activator.top! + (activator.height! - tooltip.height!) / 2) +
        +props.offsetY
      )
    })

    const computeLeftPosition = computed<number>(() => {
      return (
        (props.left
          ? activator.left! - tooltip.width!
          : props.right
            ? activator.left! + activator.width!
            : activator.left! + (activator.width! - tooltip.width!) / 2) +
        +props.offsetX
      )
    })

    const styles = computed<Record<string, string>>(() => ({
      top: tooltip.top ? convertToUnit(tooltip.top) as string : '',
      left: tooltip.top ? convertToUnit(tooltip.left) as string : '',
      maxWidth: !!props.maxWidth ? `${ props.maxWidth }px` : '',
      minWidth: !!props.minWidth ? `${ props.minWidth }px` : '',
      zIndex: props.zIndex,
    }))

    const genActivator = (): VNode | null => {
      const slotContent = renderSlot(slots, 'activator', {
        on: listeners,
      })

      return h(
        'div',
        {
          class: 'v-tooltip__activator',
          ref: activatorRef,
        },
        slotContent,
      )
    }

    const genContentDataProps = (): Record<string, any> => {
      return {
        class: contentClasses.value,
        style: styles.value,
        ref: tooltipRef,
      }
    }

    const genContent = (): VNode => {
      return withDirectives(
        h(
          'span',
          setBackground(props.color, genContentDataProps()),
          slots.default && slots.default(),
        ),
        [[vShow, innerActive.value]],
      )
    }

    const genTooltip = (): VNode => {
      const content = useTransition(
        { transition: innerActive.value ? 'scaleIn' : 'fade' },
        genContent() as VNode,
      )

      return h(
        'div',
        {
          class: classes.value,
        },
        [genActivator(), content],
      )
    }

    const setTooltipPosition = () => {
      if (tooltipRef.value) {
        tooltip.width = tooltipRef.value!.offsetWidth
        tooltip.height = tooltipRef.value!.offsetHeight
        tooltip.top = computeTopPosition.value
        tooltip.left = computeLeftPosition.value
      }
    }

    watch(
      () => isActive.value,
      to => (innerActive.value = to),
      { immediate: true },
    )

    onMounted(() => {
      watch(
        () => innerActive.value,
        to => {
          if (to) {
            const { left, top, height, width } = getActivatorSizes()

            activator.left = left as number
            activator.top = top as number
            activator.height = height as number
            activator.width = width as number

            tooltip.top = 0
            tooltip.left = 0

            requestAnimationFrame(setTooltipPosition)
          }
        },
        { immediate: true },
      )
    })

    return () => genTooltip()
  },
})
