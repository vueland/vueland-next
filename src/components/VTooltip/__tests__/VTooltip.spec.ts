import { mount, VueWrapper } from "@vue/test-utils";
import { VTooltip } from "../VTooltip";
import "regenerator-runtime/runtime";

const activatorSlot = `
        <template v-slot:activator="{ on }">
          <div class="activator" v-on="on">
          </div>
        </template>`;

const defaultSlot = "<span>some text</span>";

type MountOptions = InstanceType<typeof VTooltip>;

describe("VTooltip", () => {
  let mountFunction: (options?: any) => VueWrapper<MountOptions>;

  beforeEach(() => {
    mountFunction = (options = {}) => mount(VTooltip, options);
  });

  it("should mount component and match snapshot", () => {
    const slots = {
      activator: activatorSlot,
      default: defaultSlot,
    };
    const cmp = mountFunction({ slots });

    expect(cmp.find(".v-tooltip").attributes().style).toBe(
      "display: none;"
    );
    expect(cmp.html()).toMatchSnapshot();
  });

  it("should set activator and match snapshot", () => {
    const slots = {
      activator: activatorSlot,
      default: defaultSlot,
    };
    const cmp = mountFunction({ slots });

    expect(cmp.html()).toMatchSnapshot();
  });

  it("should trigger mouseenter event, show tooltip and match snapshot", async () => {
    const slots = {
      activator: activatorSlot,
      default: defaultSlot,
    };
    const cmp = mountFunction({ slots });

    await cmp.find(".activator").trigger("mouseenter");

    expect(cmp.find(".v-tooltip").attributes().style).toBe(undefined);
    expect(cmp.html()).toMatchSnapshot();
  });

  it("should trigger mouseleave event, hide tooltip and match snapshot", async () => {
    const slots = {
      activator: activatorSlot,
      default: defaultSlot,
    };
    const cmp = mountFunction({ slots });

    await cmp.find(".activator").trigger("mouseenter");
    await cmp.find(".activator").trigger("mouseleave");

    expect(cmp.find(".v-tooltip").attributes().style).toContain(
      "display: none;"
    );
    expect(cmp.html()).toMatchSnapshot();
  });

  it("should render component with top and match snapshot", () => {
    const slots = {
      activator: activatorSlot,
      default: defaultSlot,
    };
    const props = {
      top: true,
    };
    const cmp = mountFunction({ props, slots });

    expect(cmp.props().top).toBe(true);
    expect(cmp.find(".v-tooltip--top").exists()).toBe(true);
    expect(cmp.html()).toMatchSnapshot();
  });

  it("should render component with right and match snapshot", () => {
    const slots = {
      activator: activatorSlot,
      default: defaultSlot,
    };
    const props = {
      right: true,
    };
    const cmp = mountFunction({ props, slots });

    expect(cmp.props().right).toBe(true);
    expect(cmp.find(".v-tooltip--right").exists()).toBe(true);
    expect(cmp.html()).toMatchSnapshot();
  });

  it("should render component with bottom and match snapshot", () => {
    const slots = {
      activator: activatorSlot,
      default: defaultSlot,
    };
    const props = {
      bottom: true,
    };
    const cmp = mountFunction({ props, slots });

    expect(cmp.props().bottom).toBe(true);
    expect(cmp.find(".v-tooltip--bottom").exists()).toBe(true);
    expect(cmp.html()).toMatchSnapshot();
  });

  it("should render component with left and match snapshot", () => {
    const slots = {
      activator: activatorSlot,
      default: defaultSlot,
    };
    const props = {
      left: true,
    };
    const cmp = mountFunction({ props, slots });

    expect(cmp.props().left).toBe(true);
    expect(cmp.find(".v-tooltip--left").exists()).toBe(true);
    expect(cmp.html()).toMatchSnapshot();
  });

  it("should set z-index and match snapshot", () => {
    const slots = {
      activator: activatorSlot,
      default: defaultSlot,
    };
    const props = {
      zIndex: 100,
    };
    const cmp = mountFunction({ props, slots });

    expect(cmp.props().zIndex).toEqual(100);
    expect(cmp.find(".v-tooltip").attributes().style).toContain(
      "z-index: 100;"
    );
  });

  it("should set offsetX and match snapshot", () => {
    const slots = {
      activator: activatorSlot,
      default: defaultSlot,
    };
    const props = {
      offsetX: 100,
    };
    const cmp = mountFunction({ props, slots });
    expect(cmp.props().offsetX).toEqual(100);
    expect(cmp.html()).toMatchSnapshot();
  });

  it("should set offsetY and match snapshot", () => {
    const slots = {
      activator: activatorSlot,
      default: defaultSlot,
    };
    const props = {
      offsetY: 150,
    };
    const cmp = mountFunction({ props, slots });
    expect(cmp.props().offsetY).toEqual(150);
    expect(cmp.html()).toMatchSnapshot();
  });

  it("should set min width and match snapshot", () => {
    const slots = {
      activator: activatorSlot,
      default: defaultSlot,
    };
    const props = {
      minWidth: 150,
    };
    const cmp = mountFunction({ props, slots });

    expect(cmp.props().minWidth).toEqual(150);
    expect(cmp.find(".v-tooltip").attributes().style).toContain(
      "min-width: 150px;"
    );
    expect(cmp.html()).toMatchSnapshot();
  });

  it("should set max width and match snapshot", () => {
    const slots = {
      activator: activatorSlot,
      default: defaultSlot,
    };
    const props = {
      maxWidth: 270,
    };
    const cmp = mountFunction({ props, slots });

    expect(cmp.props().maxWidth).toEqual(270);
    expect(cmp.find(".v-tooltip").attributes().style).toContain(
      "max-width: 270px;"
    );
    expect(cmp.html()).toMatchSnapshot();
  });
});
