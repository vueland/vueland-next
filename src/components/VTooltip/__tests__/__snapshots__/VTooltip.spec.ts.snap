// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VTooltip should mount component and match snapshot 1`] = `
<div class="v-tooltip">
</div>
<transition-stub>
  <span class="v-tooltip__content grey lighten-1"
        style="display: none;"
  >
  </span>
</transition-stub>
<div class="v-tooltip__activator">
</div>
`;

exports[`VTooltip should render component with bottom and match snapshot 1`] = `
<div class="v-tooltip v-tooltip--bottom">
</div>
<transition-stub>
  <span class="v-tooltip__content grey lighten-1"
        style="display: none;"
  >
    <span>
      some text
    </span>
  </span>
</transition-stub>
<div class="v-tooltip__activator">
  <div class="activator">
  </div>
</div>
`;

exports[`VTooltip should render component with left and match snapshot 1`] = `
<div class="v-tooltip v-tooltip--left">
</div>
<transition-stub>
  <span class="v-tooltip__content grey lighten-1"
        style="display: none;"
  >
    <span>
      some text
    </span>
  </span>
</transition-stub>
<div class="v-tooltip__activator">
  <div class="activator">
  </div>
</div>
`;

exports[`VTooltip should render component with right and match snapshot 1`] = `
<div class="v-tooltip v-tooltip--right">
</div>
<transition-stub>
  <span class="v-tooltip__content grey lighten-1"
        style="display: none;"
  >
    <span>
      some text
    </span>
  </span>
</transition-stub>
<div class="v-tooltip__activator">
  <div class="activator">
  </div>
</div>
`;

exports[`VTooltip should render component with top and match snapshot 1`] = `
<div class="v-tooltip v-tooltip--top">
</div>
<transition-stub>
  <span class="v-tooltip__content grey lighten-1"
        style="display: none;"
  >
    <span>
      some text
    </span>
  </span>
</transition-stub>
<div class="v-tooltip__activator">
  <div class="activator">
  </div>
</div>
`;

exports[`VTooltip should set activator and match snapshot 1`] = `
<div class="v-tooltip">
</div>
<transition-stub>
  <span class="v-tooltip__content grey lighten-1"
        style="display: none;"
  >
    <span>
      some text
    </span>
  </span>
</transition-stub>
<div class="v-tooltip__activator">
  <div class="activator">
  </div>
</div>
`;

exports[`VTooltip should set max width and match snapshot 1`] = `
<div class="v-tooltip">
</div>
<transition-stub>
  <span class="v-tooltip__content grey lighten-1"
        style="max-width: 270px; display: none;"
  >
    <span>
      some text
    </span>
  </span>
</transition-stub>
<div class="v-tooltip__activator">
  <div class="activator">
  </div>
</div>
`;

exports[`VTooltip should set min width and match snapshot 1`] = `
<div class="v-tooltip">
</div>
<transition-stub>
  <span class="v-tooltip__content grey lighten-1"
        style="min-width: 150px; display: none;"
  >
    <span>
      some text
    </span>
  </span>
</transition-stub>
<div class="v-tooltip__activator">
  <div class="activator">
  </div>
</div>
`;

exports[`VTooltip should set offsetX and match snapshot 1`] = `
<div class="v-tooltip">
</div>
<transition-stub>
  <span class="v-tooltip__content grey lighten-1"
        style="display: none;"
  >
    <span>
      some text
    </span>
  </span>
</transition-stub>
<div class="v-tooltip__activator">
  <div class="activator">
  </div>
</div>
`;

exports[`VTooltip should set offsetY and match snapshot 1`] = `
<div class="v-tooltip">
</div>
<transition-stub>
  <span class="v-tooltip__content grey lighten-1"
        style="display: none;"
  >
    <span>
      some text
    </span>
  </span>
</transition-stub>
<div class="v-tooltip__activator">
  <div class="activator">
  </div>
</div>
`;

exports[`VTooltip should trigger mouseenter event, show tooltip and match snapshot 1`] = `
<div class="v-tooltip">
</div>
<transition-stub>
  <span class="v-tooltip__content grey lighten-1">
    <span>
      some text
    </span>
  </span>
</transition-stub>
<div class="v-tooltip__activator">
  <div class="activator">
  </div>
</div>
`;

exports[`VTooltip should trigger mouseleave event, hide tooltip and match snapshot 1`] = `
<div class="v-tooltip">
</div>
<transition-stub>
  <span class="v-tooltip__content grey lighten-1"
        style="display: none;"
  >
    <span>
      some text
    </span>
  </span>
</transition-stub>
<div class="v-tooltip__activator">
  <div class="activator">
  </div>
</div>
`;
